spec/src/main/asciidoc/chapters/language/expressions.adoc

```diff
@@ -255,7 +289,31 @@ An implementation of Jakarta Query targeting clients written in Java assigns the
 [[case-expressions]]
 ==== Case expressions
 
-**TODO**
+In the extended language, a _case expression_ has syntax given by `#general_case_expression#` or `#simple_case_expression#`.
+
+A `#general_case_expression#` has:
+
+1. a list of one or more ``#when_clause#``s, each of which has a `conditional_expression` paired with a `scalar_expression`, and
+2. a default `scalar_expression`.
+
+When a general case expression is executed, each `conditional_expression` is evaluated, in order, until one is satisfied.
+
+- If some `conditional_expression` is satisfied, then its `scalar_expression` is evaluated, and its value is the value of the whole case expression.
+- Otherwise, if no `conditional_expression` is satisfied, the default `scalar_expression` is evaluated, and its value is the value of the whole case expression.
```

Comment:
To clarify which direction the assignment of values is in. When I first read this, it sounded backward, and then I realized what it was trying to say.

```suggestion
- If some `conditional_expression` is satisfied, then its `scalar_expression` is evaluated, and the value of the whole case expression is assigned to its value.
- Otherwise, if no `conditional_expression` is satisfied, the default `scalar_expression` is evaluated, and the value of the whole case expression is assigned to its value.
```